valen-ui/src/main/java/be/twofold/valen/ui/MainView.java

```diff
@@ -13,7 +13,7 @@ public interface MainView extends View {
 
     void setAssets(List<Asset> resources);
 
-    void setupPreview(Asset asset, Object assetData);
+    void setupPreview(Asset asset, Archive archive);
```

Comment:
The order should be "parent", "child"
```suggestion
    void setupPreview(Archive archive, Asset asset);
```